tools/c7n_azure/c7n_azure/query.py

```diff
@@ -89,6 +90,34 @@ def augment(self, resources):
         return resources
 
 
+@sources.register('resource-graph')
+class ResourceGraphSource(object):
+
+    def __init__(self, manager):
+        self.manager = manager
+
+    def validate(self):
```

Comment:
do we need a minimal validation here that we have the appropriate resource type for graph client usage for the policy's resource? are all custodian resources supported by resource graph client?

Reply:
Great point. I added validation to now allow processing of resources that we cannot support. An example would be the data plane child resources such as `azure.keyvault-certificate`